packages/flutter/lib/src/widgets/pinned_header_sliver.dart

```diff
@@ -23,6 +23,20 @@ import 'framework.dart';
 /// ** See code in examples/api/lib/widgets/sliver/pinned_header_sliver.0.dart **
 /// {@end-tool}
 ///
+/// {@tool dartpad}
+/// A more elaborate example which creates an app bar that's similar to the one
+/// that appears in the iOS Settings app. In this example the pinned header
```

Comment:
Should probably point to CupertinoSliverNavigationBar as our out of the box solution once #149102 is merged. We could point to it now, but the Cupertino nav bars don't automatically go transparent yet.

Reply:
We also need an easy way to incorporate a sliver that behaves like the iOS settings search textfield: doesn't appear until the user scrolls downwards, resizes to zero when the user scrolls upwards, absorbs the upwards scroll until its extent is zero.